nautobot/project-static/js/homepage_layout.js

```diff
@@ -0,0 +1,170 @@
+                insertBefore = droppedOn;
+            } else {
+                /* Bottom half - insert after that panel */
+                insertBefore = droppedOn.nextSibling;
+            }
+        } else {
+            /* We were dropped into empty space - find the closest panel by geometry */
+            for (let child of this.children) {
+                /* Are we in the correct column? */
+                if (child.offsetLeft > e.offsetX) {
+                    /* Found the first child that is too far to the right, so we insert before that child */
+                    insertBefore = child;
+                    break;
+                } else if (child.offsetLeft + child.offsetWidth >= e.offsetX) {
+                    /* Child is in the correct column */
+                    if (child.offsetTop >= e.offsetY) {
+                        /* Found the first child in this column we were dropped above, so we insert before that child */
+                        insertBefore = child;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (insertBefore) {
+            this.insertBefore(dragged, insertBefore);
+        } else {
+            /* Add to end of the list */
+            this.append(dragged);
+        }
+
+        savePanelOrder();
+
+        // Remove the "dragging" class from the panel that was dragged
+        dragged.classList.remove("dragging");
+    }
+
+    // Function to save the order of panels in localStorage
+    function savePanelOrder() {
+        var panelOrder = Array.from(draggableHomepagePanels.children).map(function(panel) {
+            return panel.id;
+        });
+        localStorage.setItem("homepage-panels-order", JSON.stringify(panelOrder));
+    }
+
+    // Function to load the saved order of panels from localStorage
+    function loadSavedPanelOrder() {
+        var savedOrder = localStorage.getItem("homepage-panels-order");
+
+        if (savedOrder) {
+            savedOrder = JSON.parse(savedOrder);
+
+            // Append the panels in the saved order
+            for (var i = 0; i < savedOrder.length; i++) {
+                var panel = document.getElementById(savedOrder[i]);
+                if (panel) {
+                    draggableHomepagePanels.appendChild(panel);
+                }
+            }
+        }
+    }
+
+    // Enable panel dragging
+    function enablePanelDragging() {
+        var panelOrder = Array.from(draggableHomepagePanels.children).map(function(panel) {
+            return panel.id;
+        });
+        for (var i = 0; i < panelOrder.length; i++) {
+            var panel = document.getElementById(panelOrder[i]);
+            if (panel) {
+                panel.draggable = true;
+            }
+        }
+
+    }
+
+    // Fade in the panels after they have been rearranged
+    var opacity = 0; 
+    var intervalID = 0; 
+    window.onload = fadeIn; 
+      
+    function fadeIn() { 
+        setInterval(show, 50); 
```

Comment:
```suggestion
        intervalID = setInterval(show, 50); 
```

Reply:
This is needed so that the function will be stopped once the fade-in is completed. Otherwise it keeps running infinitely.